fvm/evm/types/call.go

```diff
@@ -98,13 +100,20 @@ func (dc *DirectCall) Message() *gethCore.Message {
 
 // Transaction constructs a geth.Transaction from the direct call
 func (dc *DirectCall) Transaction() *gethTypes.Transaction {
+	// Since legacyTX for a direct call doesn't have a valid siganture
+	// and we need to somehow include the From feild for the purpose
+	// of hash calculation. we define the canonical format by
+	// using the FROM bytes to set the bytes for the R part of the tx (big endian),
+	// all bytes are S is kept to zero and V is set to DirectCallTxType (255).
 	return gethTypes.NewTx(&gethTypes.LegacyTx{
 		GasPrice: big.NewInt(0),
 		Gas:      dc.GasLimit,
 		To:       dc.to(),
 		Value:    dc.Value,
 		Data:     dc.Data,
 		Nonce:    dc.Nonce,
+		V:        new(big.Int).SetBytes(dc.From.Bytes()),
+		S:        big.NewInt(255),
```

Comment:
```go
// all bytes are S is kept to zero and V is set to DirectCallTxType (255).
```
I think this line of comment also needs an update.

Reply:
Just changed the comment, also added subtype 🗡️

Reply:
Oh, nice. Subtype is the field that changes between direct calls.